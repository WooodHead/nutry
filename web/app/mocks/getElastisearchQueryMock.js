// TODO: change this structure and naming to fixtures
export const getSearchPlainSuccess = {
  timed_out: false,
  took: 10,
  _shards: {
    failed: 0,
    successful: 5,
    total: 5,
  },
  hits: {
    total: 1,
    max_score: 10,
    hits: [
      {
        _id: 'AWF8pHZ9bw9zRJtz161q',
        _index: 'nutry-names',
        _score: 10,
        _source: {
          name: 'Fresh thyme',
          group: 'Spices and Herbs',
          SN: '02049',
          alias: [],
          allergen: [],
          nutrients: {},
          portions: [],
          tags: [],
          usage: [],
        },
      },
    ],
  },
};
export const getSearchParsedSuccess = {
  items: [
    {
      _id: 'AWF8pHZ9bw9zRJtz161q',
      _index: 'nutry-names',
      _score: 10,
      _source: {
        name: 'Fresh thyme',
        group: 'Spices and Herbs',
        SN: '02049',
        alias: [],
        allergen: [],
        nutrients: {},
        portions: [],
        tags: [],
        usage: [],
      },
    },
  ],
  max_score: 10,
  hits: 1,
};
export const getProfilerQueryRequest = {
  searchString: 'chocolate',
  elements: [
    {
      nutrient: { value: 'CHOCDF', label: 'Carbohydrates', className: 'elements-carbohydrates' },
      scale: 50,
    },
    {
      nutrient: { value: 'PROCNT', label: 'Protien', className: 'elements-protein' },
      scale: 50,
    },
  ],
};
export const getProfilerPlainSuccess = {
  timed_out: false,
  took: 10,
  _shards: {
    failed: 0,
    successful: 5,
    total: 5,
  },
  hits: {
    total: 1,
    max_score: 10,
    hits: [
      {
        _id: 'AWF8pHZ9bw9zRJtz161q',
        _index: 'nutry-names',
        _score: 10,
        _source: {
          name: 'Fresh thyme',
          group: 'Spices and Herbs',
          SN: '02049',
          alias: [],
          allergen: [],
          nutrients: {},
          portions: [],
          tags: [],
          usage: [],
        },
      },
    ],
  },
};
export const getProfilerParsedSuccess = {
  items: [
    {
      _id: 'AWF8pHZ9bw9zRJtz161q',
      _index: 'nutry-names',
      _score: 10,
      _source: {
        name: 'Fresh thyme',
        group: 'Spices and Herbs',
        SN: '02049',
        alias: [],
        allergen: [],
        nutrients: {},
        portions: [],
        tags: [],
        usage: [],
      },
    },
  ],
  max_score: 10,
  hits: 1,
};
export const getDocumentPlainSuccess = {
  timed_out: false,
  took: 10,
  _shards: {
    failed: 0,
    successful: 5,
    total: 5,
  },
  hits: {
    total: 1,
    max_score: 10,
    hits: [
      {
        _id: 'AWF8pHZ9bw9zRJtz161q',
        _index: 'nutry-names',
        _score: 10,
        _source: {
          name: 'Fresh thyme',
          group: 'Spices and Herbs',
          SN: '02049',
          alias: [],
          allergen: [],
          nutrients: {},
          portions: [],
          tags: [],
          usage: [],
        },
      },
    ],
  },
};
export const getDocumentParsedSuccess = {
  items: [
    {
      _id: 'AWF8pHZ9bw9zRJtz161q',
      _index: 'nutry-names',
      _score: 10,
      _source: {
        name: 'Fresh thyme',
        group: 'Spices and Herbs',
        SN: '02049',
        alias: [],
        allergen: [],
        nutrients: {},
        portions: [],
        tags: [],
        usage: [],
      },
    },
  ],
  max_score: 10,
  hits: 1,
};

export const searchQueryMock = {
  bool: {
    should: [{
      match_phrase_prefix: {
        name: {
          query: 'chocolate',
          max_expansions: 10,
          slop: 10,
        },
      },
    },
    {
      match: {
        name: {
          query: 'chocolate',
          fuzziness: 'AUTO',
          operator: 'and',
          prefix_length: 1,
        },
      },
    }],
  },
};
export const profilerQueryMock = {
  bool: {
    boost: 0.001,
    should: [{
      match_phrase_prefix: {
        name: {
          query: 'chocolate',
          max_expansions: 50,
          slop: 50,
        },
      },
    },
    {
      match: {
        name: {
          query: 'chocolate',
          fuzziness: 'AUTO',
          operator: 'and',
          prefix_length: 1,
        },
      },
    }],
  },
};
export const profilerFunctionQueryMock = {
  field_value_factor: {
    field: 'CHOCDF',
    factor: 10,
    missing: 1,
  },
};
export const profilerDecayFunctionQueryMock = {
  gauss: {
    CHOCDF: {
      origin: 10,
      scale: 200,
    },
  },
};
export const matchFieldMock = {
  match: {
    SN: '02049',
  },
};
