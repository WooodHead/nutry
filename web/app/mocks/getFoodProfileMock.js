export const getFoodProfileMockSuccess = {
  status: 200,
  statusText: 'OK',
  response: {
    name: 'Fresh thyme',
    group: 'Spices and Herbs',
    SN: '02049',
    alias: [],
    allergen: [],
    nutrients: {},
    portions: [],
    tags: [],
    usage: [],
  },
};
export const getFoodProfileMockSuccessFull = {
  status: 200,
  statusText: 'OK',
  response: {
    name: 'Fresh thyme',
    group: 'Spices and Herbs',
    SN: '02049',
    alias: [],
    allergen: [],
    nutrients: {
      CHOCDF: {
        name: 'Carbohydrate, by difference',
        units: 'g',
        value: 1 },
      ALC: {
        name: 'Alcohol, ethyl',
        units: 'g',
        value: 2 },
      ARGN: {
        name: 'Arginine',
        units: 'g',
        value: 0.803 },
    },
    portions: [],
    tags: [],
    usage: [],
  },
};

export const getMultiFoodProfileMockSuccess = [
  {
    status: 200,
    statusText: 'OK',
    response: {
      name: 'item1',
      group: 'group1',
      SN: '00000',
      alias: [],
      allergen: [],
      nutrients: {},
      portions: [],
      tags: [],
      usage: [],
    },
  },
  {
    status: 200,
    statusText: 'OK',
    response: {
      name: 'item2',
      group: 'group2',
      SN: '00001',
      alias: [],
      allergen: [],
      nutrients: {},
      portions: [],
      tags: [],
      usage: [],
    },
  },
];
