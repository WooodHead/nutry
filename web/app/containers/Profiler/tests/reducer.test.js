import { getProfilerParsedSuccess } from 'mocks/getElasticsearchQueryMock';
import {
    addProfilerElement,
    searchComplete,
    searchLoading,
    searchFailure,
    deleteProfilerElement,
    updateProfilerElement,
    changeSearch } from '../actions';

import profilerReducer from '../reducer';

describe('Container [Profiler] - reducers', () => {
  let state;
  beforeEach(() => {
  });
  it('should return the initial state', () => {
    expect(profilerReducer(undefined, {})).toMatchSnapshot();
  });
  it('should handle the changeSearch action correctly', () => {
    expect(profilerReducer(state, changeSearch())).toMatchSnapshot();
  });
  it('should handle the searchComplete action correctly', () => {
    expect(profilerReducer(state, searchComplete(getProfilerParsedSuccess))).toMatchSnapshot();
  });
  it('should handle the addProfilerElement action correctly', () => {
    expect(profilerReducer(state, addProfilerElement()).get('nutrient')).toMatchSnapshot();
  });
  it('should handle the deleteProfilerElement action correctly', () => {
    expect(profilerReducer(state, deleteProfilerElement('uuid'))).toMatchSnapshot();
  });
  it('should handle the updateProfilerElement action correctly', () => {
    expect(profilerReducer(state, updateProfilerElement('uuid'))).toMatchSnapshot();
  });
  it('should handle the loadProfileSuccess action correctly', () => {
    expect(profilerReducer(state, searchLoading())).toMatchSnapshot();
  });
  it('should handle loadProfileFailure correctly', () => {
    const err = new Error('failed to retrieve profile');
    expect(profilerReducer(state, searchFailure(err))).toMatchSnapshot();
  });
});
