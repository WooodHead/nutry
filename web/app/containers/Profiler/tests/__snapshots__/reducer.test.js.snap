// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [Profiler] - reducers should handle loadProfileFailure correctly 1`] = `
Immutable.Map {
  "loading": false,
  "results": Immutable.List [],
  "elements": Immutable.Map {},
}
`;

exports[`Container [Profiler] - reducers should handle the addProfilerElement action correctly 1`] = `
Immutable.Map {
  "loading": false,
  "results": Immutable.List [],
  "elements": Immutable.Map {
    "fcf1d884-ed6e-4378-bafc-8a9948cf69c5": Immutable.Map {
      "nutrient": Object {
        "className": "elements-carbohydrates",
        "label": "Carbohydrates",
        "value": "CHOCDF",
      },
      "scale": 50,
    },
  },
}
`;

exports[`Container [Profiler] - reducers should handle the changeSearch action correctly 1`] = `
Immutable.Map {
  "loading": true,
  "results": Immutable.List [],
  "elements": Immutable.Map {},
}
`;

exports[`Container [Profiler] - reducers should handle the deleteProfilerElement action correctly 1`] = `
Immutable.Map {
  "loading": false,
  "results": Immutable.List [],
  "elements": Immutable.Map {},
}
`;

exports[`Container [Profiler] - reducers should handle the loadProfileSuccess action correctly 1`] = `
Immutable.Map {
  "loading": false,
  "results": Immutable.List [],
  "elements": Immutable.Map {},
}
`;

exports[`Container [Profiler] - reducers should handle the searchComplete action correctly 1`] = `
Immutable.Map {
  "loading": false,
  "results": Object {
    "hits": 1,
    "items": Array [
      Object {
        "_id": "AWF8pHZ9bw9zRJtz161q",
        "_index": "nutry-names",
        "_score": 10,
        "_source": Object {
          "SN": "02049",
          "alias": Array [],
          "allergen": Array [],
          "group": "Spices and Herbs",
          "name": "Fresh thyme",
          "nutrients": Object {},
          "portions": Array [],
          "tags": Array [],
          "usage": Array [],
        },
      },
    ],
    "max_score": 10,
  },
  "elements": Immutable.Map {},
}
`;

exports[`Container [Profiler] - reducers should handle the updateProfilerElement action correctly 1`] = `
Immutable.Map {
  "loading": false,
  "results": Immutable.List [],
  "elements": Immutable.Map {
    "uuid": Immutable.Map {
      undefined: undefined,
    },
  },
}
`;

exports[`Container [Profiler] - reducers should return the initial state 1`] = `
Immutable.Map {
  "loading": false,
  "results": Immutable.List [],
  "elements": Immutable.Map {},
}
`;
